dinky-admin/src/main/java/org/dinky/controller/BaseController.java

```diff
@@ -0,0 +1,81 @@
+package org.dinky.controller;
+
+import org.dinky.common.result.Result;
+import org.dinky.constant.BaseConstant;
+import org.dinky.enums.Status;
+
+import java.util.Map;
+
+public class BaseController {
+    /**
+     * return data list
+     *
+     * @param result result code
+     * @return result code
+     */
+    public Result returnDataList(Map<String, Object> result) {
+        Status status = (Status) result.get(BaseConstant.STATUS);
+        if (status == Status.SUCCESS) {
+            String msg = Status.SUCCESS.getMsg();
+            Object datalist = result.get(BaseConstant.DATA_LIST);
+            return success(msg, datalist);
+        } else {
+            Integer code = status.getCode();
+            String msg = (String) result.get(BaseConstant.MSG);
+            return error(code, msg);
+        }
+    }
+
+    /**
+     * success handle
+     *
+     * @param msg success message
+     * @param list data list
+     * @return success result code
+     */
+    public Result success(String msg, Object list) {
```

Comment:
these all methods could be static, and Result be generic format as all your other improve. `public static T Result<T> success(String msg, T list) {`

Reply:
You're right, I'm considering not using generics here. The purpose is to make the code clearer because the returned type is unknown. If you use generics, there may be a place to modify it later, and it doesn't feel good to have to modify all the references, and the code's aesthetics are also not very high.

Reply:
what about the "a place to modify it"?

Reply:
![2d073ce6da004a1c7ba541f37fa9112](https://github.com/DataLinkDC/dinky/assets/63942121/eb2e9c6c-5167-4b6a-8d45-701bcd57660b)
Is it really elegant to add a generic to the returned results like this?

Reply:
[Effective Java - 3rd Edition- Item 26](https://ahdak.github.io/blog/effective-java-part-4/), explain why shouldn't use raw type.

Reply:
Niubi, I will make changes according to your requirements later.

Reply:
Thank you very much. You are right and have made the modifications according to your requirements. Let's add a method to BaseController first. If you feel that this is possible, I will have time to modify other controllers in this format later.